plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/util/CustomMessageConsole.java

```diff
@@ -0,0 +1,128 @@
+/*******************************************************************************
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * All rights reserved. This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package com.google.cloud.tools.eclipse.util;
+
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.action.IToolBarManager;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.ui.console.IConsole;
+import org.eclipse.ui.console.IConsoleConstants;
+import org.eclipse.ui.console.IConsolePageParticipant;
+import org.eclipse.ui.console.MessageConsole;
+import org.eclipse.ui.part.IPageBookViewPage;
+
+/**
+ * A message console with additional features, such as displaying a terminate
+ * button.
+ */
+public class CustomMessageConsole extends MessageConsole {
```

Comment:
It is referred by the deploy code (that is broken right now, see #66 )
I'd rather keep it around until the deploy is implemented properly and getting rid of it then if it is not needed, but open to delete it and all deploy related code. WDYT @briandealwis @elharo ?